disperser/dataapi/ejector.go

```diff
@@ -0,0 +1,124 @@
+package dataapi
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// The caller should ensure "stakeShare" is in range (0, 1].
+func stakeShareToSLA(stakeShare float64) float64 {
+	switch {
+	case stakeShare > 0.1:
+		return 0.975
+	case stakeShare > 0.05:
+		return 0.95
+	default:
+		return 0.9
+	}
+}
+
+// operatorPerfScore scores an operator based on its stake share and nonsigning rate. The
+// performance score will be in range [0, 1], with higher score indicating better performance.
+func operatorPerfScore(stakeShare float64, nonsigningRate float64) float64 {
+	if nonsigningRate == 0 {
+		return 1.0
+	}
+	sla := stakeShareToSLA(stakeShare)
+	perf := (1 - sla) / nonsigningRate
+	return perf / (1.0 + perf)
```

Comment:
I'm a bit uncertain about whether this scoring will give us the right ordering across SLA classes. Interested to know if there was a rationale for this particular scoring. I'll put some further thought into it.